### PR TITLE
Add Pool Type Query

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #2739 

## What is the purpose of the change

Adds pool type query to query the type of pool. The query would return a string of the type of pool in a string, depending on the type of pool, would return an error otherwise.

## Brief Changelog
- Add pool type query 

## Testing and Verifying
Added new test cases 
## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes 
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?yes
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/keeper/grpc_query.go [hunk 1]
```diff
@@ -127,6 +127,27 @@ func (q Querier) NumPools(ctx context.Context, _ *types.QueryNumPoolsRequest) (*
...
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	pool, err := q.Keeper.GetPoolAndPoke(sdkCtx, req.PoolId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	switch pool := pool.(type) {
+	case *balancer.Pool:
+		return &types.QueryPoolTypeResponse{PoolType: "Balancer"}, nil
```
Comment (ValarDragon) [inline x/gamm/keeper/grpc_query.go:144]: We should have this be exposed as a method on the pool interface, right?

Or at least, lets make a go API for `keeper.PoolType(ctx, poolId) (string, error)`, to have it in a more discoverable point

Comment (mattverse) [reply x/gamm/keeper/grpc_query.go]: Changed to having go API, I would prefer having further discussion in adding pool type to `PoolI`, I'd like to go ahead and open an issue for this for further discussion for this if you agree!

Comment (ValarDragon) [reply x/gamm/keeper/grpc_query.go]: SGTM

**Review by ValarDragon — APPROVED** [review]
Perfect!

Can you make a followup issue for discussing adding it to the poolI api?

**Comment by mattverse** [comment]
https://github.com/osmosis-labs/osmosis/issues/2868

**Pull request merged**